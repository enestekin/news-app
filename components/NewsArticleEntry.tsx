import { NewsArticle } from "@/models/NewsArticles";
import Image from "next/image";
import React from "react";
import { Card } from "react-bootstrap";
import placeholderImage from "@/assets/images/newsarticle_placeholder.jpg";
import styles from "@/styles/NewsArticleEntry.module.css";

interface NewsArticleEntryProps {
  article: NewsArticle;
}

const NewsArticleEntry: React.FC<NewsArticleEntryProps> = ({
  article: { title, description, url, urlToImage },
}) => {
  const validImageUrl =
    urlToImage?.startsWith("http://") || urlToImage?.startsWith("https://")
      ? urlToImage
      : undefined;

  return (
    <a href={url}>
      <Card className="h-100">
        <Image
          src={validImageUrl || placeholderImage}
          width={500}
          height={200}
          alt="image"
          className={`card-img-top ${styles.image}`}
        />
        <Card.Body>
          <Card.Title>{title}</Card.Title>
          <Card.Text>{description}</Card.Text>
        </Card.Body>
      </Card>
    </a>
  );
};

export default NewsArticleEntry;
