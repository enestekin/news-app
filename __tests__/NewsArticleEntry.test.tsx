import { render } from "@testing-library/react";
import NewsArticleEntry from "@/components/NewsArticleEntry";

describe("NewsArticleEntry Component", () => {
  const mockArticle = {
    title: "Test article",
    description: "This is a test description",
    url: "https://testarticle.com",
    urlToImage: "https://testimage.com",
    author: "Enes",
    publishedAt: "some date",
    content: "some content",
  };

  // it("should render the news article title", () => {
  //   render(<NewsArticleEntry article={mockArticle} />);
  //   const titleElement = screen.getByText(/Test article/i);
  //   expect(titleElement).toBeInTheDocument();
  // });

  // it("should render the news article description", () => {
  //   render(<NewsArticleEntry article={mockArticle} />);
  //   const descriptionElement = screen.getByText(/This is a test description/i);
  //   expect(descriptionElement).toBeInTheDocument();
  // });

  // it("should render the news article image", () => {
  //   render(<NewsArticleEntry article={mockArticle} />);
  //   const imageElement = screen.getByAltText(/image/i);
  //   expect(imageElement).toBeInTheDocument();
  // });

  // it("should render the news article link", () => {
  //   render(<NewsArticleEntry article={mockArticle} />);
  //   const linkElement = screen.getByTestId("link-element");
  //   expect(linkElement).toHaveAttribute("href", "https://testarticle.com");
  // });

  it("renders correctly", () => {
    const { container } = render(<NewsArticleEntry article={mockArticle} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
