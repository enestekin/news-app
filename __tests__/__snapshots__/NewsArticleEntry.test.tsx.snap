// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticleEntry Component renders correctly 1`] = `
<a
  data-testid="link-element"
  href="https://testarticle.com"
>
  <div
    class="h-100 card"
  >
    <img
      alt="image"
      class="card-img-top image"
      data-nimg="1"
      decoding="async"
      height="200"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Ftestimage.com&w=1080&q=75"
      srcset="/_next/image?url=https%3A%2F%2Ftestimage.com&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Ftestimage.com&w=1080&q=75 2x"
      style="color: transparent;"
      width="500"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Test article
      </div>
      <p
        class="card-text"
      >
        This is a test description
      </p>
    </div>
  </div>
</a>
`;
